lib/logstash/outputs/xmpp.rb

```diff
@@ -51,6 +51,7 @@ def register
   public
   def connect
     Jabber::debug = true
+    Thread::abort_on_exception = false
```

Comment:
this is a global setting and it would cause unpredictable behaviour in logstash if we'd merge it.
I think the easiest solution is to move most of the `register` method content into the `run` method. If run has an exception the pipeline will restart the plugin by running `run` again, so we should probably move the connect part to it.

This means moving the following to `run`:

```ruby
  @client = connect

    @mucs = []
    @users = [] if !@users

    # load the MUC Client if we are joining rooms.
    if @rooms && !@rooms.empty?
      require 'xmpp4r/muc'
      @rooms.each do |room| # handle muc messages in different rooms
        muc = Jabber::MUC::MUCClient.new(@client)
        muc.join(room)
        @mucs << muc
      end # @rooms.each
    end # if @rooms
```

Also, I think all code related to `@mucs` , `@users` and `@rooms`  could be encapsulated into a `setup_muc` method. This way the `run` method executes `connect` and `setup_muc`.
What do you think?

Reply:
Sorry for the late response, crazy week!

OK I'll have a look at this, I've never done any hacking on Logstash / Plugins before and I'm more of an ops guy than a dev guy (mostly write Puppet) so I'll have a go and get back to you :)

Thanks for the constructive advice, it's really appreciated.

Reply:
Oh.my.god. jruby... is a PITA 😡

Reply:
Yeah I had a go but I don't really know what I'm doing TBH.